### PR TITLE
More strict ts: generated types

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] before all of the common-adapters were pulled in from the constants/types side due to these icon.constants to get IconType, instead just pull in this type and not everything

### CODE REVIEW

File: shared/constants/types/fs.tsx [hunk 1]
```diff
@@ -632,7 +632,7 @@ export const stringToPath = (s: string): Path =>
 export const pathToString = (p: Path): string => (!p ? '' : p)
 export const stringToLocalPath = (s: string): LocalPath => s
 export const localPathToString = (p: LocalPath): string => p
-export const getPathName = (p: Path): string => (!p ? '' : p.split('/').pop())
+export const getPathName = (p: Path): string => (!p ? '' : p.split('/').pop() || '')
```
Comment (chrisnojima) [inline shared/constants/types/fs.tsx]: @songgao please sanity check

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: shared/constants/types/fs.tsx (hunk 1 above)
Comment (songgao) [reply shared/constants/types/fs.tsx]: This is sane ✅

File: shared/constants/types/more.tsx [hunk 2]
```diff
@@ -1,56 +1,64 @@
...
+  | 'facebook'
+  | 'github'
+  | 'hackernews'
+  | 'http'
+  | 'https'
+  | 'reddit'
+  | 'rooter'
+  | 'twitter'
+  | 'web'
+  | 'rooter'
```
Comment (songgao) [inline shared/constants/types/more.tsx]: Any reason why we didn't go with enum here? Not suggesting a change necessarily; just curious

File: shared/actions/profile/proofs.tsx [hunk 3]
```diff
@@ -61,7 +61,7 @@ const recheckProof = (state, action: ProfileGen.RecheckProofPayload) =>
 // only let one of these happen at a time
 let addProofInProgress = false
 function* addProof(state, action: ProfileGen.AddProofPayload) {
-  const service = More.isPlatformsExpandedType(action.payload.platform)
+  const service = More.asPlatformsExpandedType(action.payload.platform)
```
Comment (songgao) [inline shared/actions/profile/proofs.tsx]: This can still be the `is` version right?

Comment (chrisnojima) [reply shared/actions/profile/proofs.tsx]: no, the type guard wants a boolean. this original function is a cast to the thing or null, so i renamed the old on `as` and made the type guard `is` as i think its more accurate

Comment (songgao) [reply shared/actions/profile/proofs.tsx]: Oh right. I missed the `switch (service)` below and thought it was only used in the line immediately below. LGTM

**Pull request merged**